src/main/kotlin/com/eodigo/batch/controller/BatchJobController.kt

```
@@ -40,19 +46,36 @@ class BatchJobController(
+                        !isIsoLocalDate(surveyDate) -> throw InvalidInputValueException()
+                        else -> surveyDate
+                    }
+                jobParametersBuilder.addString("surveyDate", finalSurveyDate)
             }
 
             val jobParameters = jobParametersBuilder.toJobParameters()
 
             jobLauncher.run(job, jobParameters)
 
             ResponseEntity.ok("Batch job '$jobName' has been started.")
+        } catch (e: InvalidInputValueException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)